core/src/commonMain/kotlin/com/ustadmobile/core/domain/validateusername/ValidateUsernameUseCase.kt

```diff
@@ -0,0 +1,69 @@
+package com.ustadmobile.core.domain.validateusername
+
+class ValidateUsernameUseCase {
+    /**
+     * Validates username according to rules:
+     * - Must not contain special characters (except . and _)
+     * - Must not start with a number
+     * - All letters will be converted to lowercase
+     * - Returns cleaned username or null if invalid
+     *
+     * @param username The username string to validate
+     * @param invalidReplacement String to replace invalid characters with (default empty string)
+     * @return The valid username if valid, or null if invalid
+     */
+    operator fun invoke(
+        username: String,
+        invalidReplacement: String = ""
+    ): String? {
+        val trimmed = username.trim()
+        if (trimmed.isEmpty()) return null
+
+        // Convert to lowercase
+        val lowercased = trimmed.lowercase()
+
+        // Check if starts with number
+        if (lowercased.firstOrNull()?.isDigit() == true) {
+            return null
+        }
+
+        // Check for invalid characters
+        val containsInvalidChars = lowercased.any { !isCharacterAllowed(it) }
+        if (containsInvalidChars && invalidReplacement.isEmpty()) {
+            return null
+        }
+
+        // Replace invalid characters if replacement provided
+        return if (containsInvalidChars) {
+            lowercased.map { char ->
+                when {
+                    !isCharacterAllowed(char) -> invalidReplacement
```

Comment:
What's going on here is two separate logic tracks - one dealing with white space, and another other non-allowed characters? Why not just have isCharacterAllowed return false when character is white space? Why have different logic handling white space vs. other non-allowed characters?